infra/mock_api/template/ecs.tf

```diff
@@ -109,21 +106,132 @@ resource "aws_ecs_task_definition" "mock-api-ecs-task-definition" {
+          "name" : "DB_HOST"
+          "value" : "test-wic-mt.cfsgsglh2mrn.us-east-1.rds.amazonaws.com"
+        }
+      ]
       logConfiguration = {
         logDriver = "awslogs",
         options = {
-          "awslogs-group"  = "${data.aws_cloudwatch_log_group.mock_api}"
-          "awslogs-region" = "us-east-1"
+          "awslogs-group"         = "mock-api",
+          "awslogs-region"        = "us-east-1",
+          "awslogs-stream-prefix" = "mock-api"
         }
       }
-      readonlyRootFilesystem = true
+    }
+  ])
+}
+# ------------------------------------------------------------------------------
+#    ECS task to Handle CSVs
+#
+# ------------------------------------------------------------------------------
+# resource "aws_iam_role" "handle-csv" {
+#   name = "handle-csv-role"
+#   description = "allows an ECS task to generate CSVs and manage their storage"
+#   policy = data.aws_iam_policy_document.handle-csv
+# }
+# resource "aws_iam_role_policy" "handle-csv" {
+# add s3 perms (read, write, list)
+# read rds?
+# }
+
+resource "aws_security_group" "handle-csv" {
+  description = "allows internal connections"
+  name        = "csv-handler"
+  vpc_id      = module.constants.vpc_id
+
+  ingress {
+    from_port       = 5432
+    to_port         = 5432
+    protocol        = "tcp"
+    security_groups = ["${aws_security_group.allow-api-traffic.id}"]
+  }
+  egress {
+    description      = "allow all outbound traffic"
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+}
+resource "aws_ecs_task_definition" "handle-csv" {
+  family                   = "${var.environment_name}-csv-handler-definition"
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  memory                   = "1024"
+  cpu                      = "512"
+  # task_role_arn            =  aws_iam_role.handle-csv.arn
+  execution_role_arn = "arn:aws:iam::546642427916:role/wic-mt-task-executor"
+  container_definitions = jsonencode([
+    {
+      name      = "${var.environment_name}-mock-api-container"
+      image     = "546642427916.dkr.ecr.us-east-1.amazonaws.com/mock-api-repo:latest"
+      memory    = 1024
+      cpu       = 512
+      essential = true
+      command   = ["poetry", "run", "create-eligibility-screener-csv"]
+      portMappings = [
+        {
+          containerPort : 8080
+        }
+      ]
       linuxParameters = {
         capabilities = {
           drop = ["ALL"]
         },
         initProcessEnabled = true
       }
+      environment : [
+        {
+          "name" : "POSTGRES_USER"
+          "value" : "${data.aws_ssm_parameter.db_username.value}"
+        },
+        {
+          "name" : "POSTGRES_PASSWORD"
+          "value" : "${data.aws_ssm_parameter.db_pw.value}"
+        },
+        {
+          "name" : "POSTGRES_DB"
+          "value" : "${data.aws_ssm_parameter.db.value}"
+        },
```

Comment:
generally for SSM secrets you'll want to use `valueFrom` and provide the SSM path directly. That way you don't need a full `terraform apply` to update a secret, and it'll prevent those secret values from potentially showing up in plaintext in the tfstate (although I'm not sure if that still happens in the latest terraform/provider versions.)

`"valueFrom": "POSTGRES_DB"`

Reply:
(I also recognize you need to pull in the master username/password anyways to create the RDS instance in the first place 😂 🤷 so not super important right now)